src/WpfMath/TexPredefinedFormulaParser.cs

```diff
@@ -315,24 +314,33 @@ public override object Parse(string value, string type)
 
         public abstract class ActionParser : ParserBase
         {
+            protected  ActionParser(IDictionary<string, TexFormula> sharedCacheFormulas)
+                : base(sharedCacheFormulas)
+            {}
+
             public abstract void Parse(SourceSpan source, XElement element);
         }
 
         public abstract class ArgumentValueParser : ParserBase
         {
+            protected ArgumentValueParser(IDictionary<string, TexFormula> sharedCacheFormulas)
+                : base(sharedCacheFormulas)
+            {}
+
             public abstract object Parse(string value, string type);
         }
 
         public abstract class ParserBase
         {
-            public ParserBase()
+            public ParserBase(IDictionary<string, TexFormula> sharedCacheFormulas)
             {
+                this.SharedCacheFormulas = sharedCacheFormulas;
             }
 
-            public IDictionary<string, TexFormula> TempFormulas
+            public IDictionary<string, TexFormula> SharedCacheFormulas
             {
                 get;
-                set;
+                private set;
```

Comment:
The whole setter is actually unnecessary. Get-only properties are settable in constructors anyway.